Rework hardware page (#241)
micro-ROS/micro-ROS.github.io: _docs/overview/hardware/index.md

permalink: /docs/overview/hardware/
---

<style>
.hardwarecontainer {
  height: 300px;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.hardwareitem_image {
  width: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.hardwareitem_description {
  width: 50%;  
}

.hardwareitem_image img {
    max-width: 80%;
}
</style>

Micro-ROS aims to **bring ROS 2 to a wide set of microcontrollers** to allow having first-class ROS 2 entities in the embedded world.

> The main targets of micro-ROS are mid-range 32-bits microcontroller families. Usually, the minimum requirements for running micro-ROS in an embedded platform are memory constrains. Since memory usage in micro-ROS is a complex matter we have a [complete article](https://micro-ros.github.io/docs/concepts/memo_prof/) describing it and a tutorial on [how to tune the memory consuption](https://micro-ros.github.io/docs/tutorials/core/microxrcedds_rmw_configuration/) in the micro-ROS middleware. 

Reviewer: ```suggestion
The main targets of micro-ROS are mid-range 32-bits microcontroller families. Usually, the minimum requirements for running micro-ROS in an embedded platform are memory constrains. Since memory usage in micro-ROS is a complex matter we provide a [complete article](https://micro-ros.github.io/docs/concepts/memo_prof/) describing it and a tutorial on [how to tune the memory consuption](https://micro-ros.github.io/docs/tutorials/core/microxrcedds_rmw_configuration/) in the micro-ROS middleware. 
```

> The main targets of micro-ROS are mid-range 32-bits microcontroller families. Usually, the minimum requirements for running micro-ROS in an embedded platform are memory constrains. Since memory usage in micro-ROS is a complex matter we have a [complete article](https://micro-ros.github.io/docs/concepts/memo_prof/) describing it and a tutorial on [how to tune the memory consuption](https://micro-ros.github.io/docs/tutorials/core/microxrcedds_rmw_configuration/) in the micro-ROS middleware. 

Reviewer 2: ```suggestion
The main targets of micro-ROS are mid-range 32-bits microcontroller families. Usually, the minimum requirements for running micro-ROS in an embedded platform are memory constraints. Since memory usage in micro-ROS is a complex matter we have a [complete article](https://micro-ros.github.io/docs/concepts/memo_prof/) describing it and a tutorial on [how to tune the memory consuption](https://micro-ros.github.io/docs/tutorials/core/microxrcedds_rmw_configuration/) in the micro-ROS middleware. 
```

> In general micro-ROS will need MCU that have tens of kilobytes of RAM memory and communication peripherals that enables the micro-ROS [client to agent communication](https://micro-ros.github.io//docs/overview/features/).

Reviewer: ```suggestion
In general micro-ROS will need MCUs that have tens of kilobytes of RAM memory and communication peripherals that enable the micro-ROS [Client to Agent communication](https://micro-ros.github.io//docs/overview/features/).
```

> The micro-ROS hardware support is divided in three categories: 

Reviewer 2: ```suggestion
The micro-ROS hardware support is divided into three categories: 
```

- reference micro-ROS board, 
- tier 2 boards and, 
- community supported boards.

> *In order to check the most recent hardware support visit the [micro_ros_setup repo](https://github.com/micro-ROS/micro_ros_setup)*

Reviewer: ```suggestion
*In order to check the most recent hardware support visit the [micro_ros_setup repo](https://github.com/micro-ROS/micro_ros_setup)*.
```

## Reference micro-ROS board

> The micro-ROS reference boards are the one officially supported for all RTOS with complete support for all available transports.

Reviewer: ```suggestion
The micro-ROS reference boards are the ones officially supported for all RTOSes and with complete support for all available transports.
```

<div class="hardwarecontainer">
  <div class="hardwareitem_description">
    <h3><b>Olimex LTD STM32-E407</b></h3>
    <div>
        <b>Key features:</b>
        <ul>
            <li>MCU: STM32F407ZGT6 Cortex-M4F</li>
            <li>RAM: 196 kB</li>
            <li>Flash: 1 MB</li>
            <li>Peripherals:  USB OTG, Ethernet, SD Card slot, SPI, CAN, I2C... </li>
        </ul>  
        
        <b>Resources:</b>
        <ul>
            <li><a href="https://www.olimex.com/Products/ARM/ST/STM32-E407/open-source-hardware">Official website</a></li>
            <li><a href="https://github.com/OLIMEX/STM32F4/blob/master/HARDWARE/STM32-E407/STM32-E407_Rev_F.pdf">Schematics</a></li>
            <li><a href="https://www.olimex.com/Products/ARM/ST/STM32-E407/resources/STM32-E407.pdf">User Manual</a></li>
        </ul>    
    </div>
  </div>

  <div class="hardwareitem_image">
    <img src="https://www.olimex.com/Products/ARM/ST/STM32-E407/images/STM32-E407-02.jpg">
  </div>
</div>

## Tier 2 micro-ROS boards

> The micro-ROS Tier 2 boards are officially suported boards for one or more RTOS or transports.

Reviewer: ```suggestion
The micro-ROS Tier 2 boards are officially supported for one or more RTOSes and transports.
```
